features/steps/steps.py

```diff
@@ -173,6 +189,42 @@ def step_impl(context, attribute, value):
         filter(lambda inst: getattr(inst, attribute) == value, context.instances)
     )
 
+def add_attribute_values(i, attribute, context):
+    if context.get('attribute'):
+        return (i[0], getattr(i[1], attribute, None))
+    else:
+        return (i, getattr(i, attribute, None))
+
+@given('Its values for attribute {attribute}')
+def step_impl(context, attribute):
+    values = list(map(lambda i: (i, getattr(i, attribute, None)), context.instances))
+    filter_nones = list(filter(lambda i: i[1] is not None, values))
+
+    context._push()
+    context.instances = []
+    for var, attr in filter_nones: 
+        setattr(context, 'entities', {'var':var, attribute:attr})
+        context.instances.append(attr)
+    setattr(context, 'attribute', attribute)
+    
+
+@given('The element has {constraint} {num:d} instance(s) of {entity}')
+def step_impl(context, constraint, num, entity):
+    ent_attr = {'IfcShapeRepresentation':'Representations'}
+    assert entity in ent_attr
+    attr = ent_attr[entity]
+
+    stmt_to_op = {"at least": operator.ge, "more than": operator.gt}
+    assert constraint in stmt_to_op
+    op = stmt_to_op[constraint]
```

Comment:
Can we have a specialized function that resolves these? I feel like I've seen a construct like this too many times already ...